### PR TITLE
zx spectrum drivers - restore irq length == 32 / border timings

### PR DESCRIPTION
**Pull request by holub**
zx spectrum drivers - restore irq length 32
more precise timings for screen and border updates
simplify current frame shift calculation which improves performance
minor: some states init/save

### CODE REVIEW

**Comment by holub** [comment]
border updates more precise now

before:
fpga48_aftter

after:
fpga48_after

File: src/mame/drivers/pentagon.cpp [hunk 1]
```diff
@@ -217,8 +217,11 @@ void pentagon_state::machine_reset()
 
 void pentagon_state::video_start()
 {
-	m_frame_invert_count = 16;
-	m_screen_location = m_ram->pointer() + (5 << 14);
+	spectrum_128_state::video_start();
+	m_contention_pattern = {};
+	/* XXX Must be 4 but that requires draw of previous border chunk and significantly complicates the code.
+	Will see if we really have case for this change. */
```
Comment (cuavas) [inline src/mame/drivers/pentagon.cpp:223]: Please use `TODO` or `FIXME` to indicate severity of things like this (they’re also picked up by editors like vim and several static analysis tools), not `XXX`.

**Comment by holub** [comment]
finally Shock works on +3
shock_plus3

**Review by cuavas — COMMENTED** [review]
Are the resident ZX Spectrum experts happy with this?  The code holds up to a “checklist review” now.

**Comment by DavidHaywood** [comment]
I'd consider holub to be the resident ZX Spectrum expert at this point.

As long as it passes standards then I'm fine with it, and if any issues come up, they'll be reported back.

**Comment by holub** [comment]
> I'd consider holub to be the resident ZX Spectrum expert at this point.
> 
> As long as it passes standards then I'm fine with it, and if any issues come up, they'll be reported back.

Appreciate your comment!
Going forward please keep eye on M.A.M.E. core usage in my PRs, the area I'm not fully comfortable with yet.

**Pull request merged**